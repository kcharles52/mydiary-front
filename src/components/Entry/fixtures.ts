export const props = {
  entry_id: 1,
  title: "Title 1",
  diaryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete:jest.fn()

};
