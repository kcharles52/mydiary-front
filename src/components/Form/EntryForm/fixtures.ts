export const EntryFormProps = {
  Edit: false,
  title: "string",
  diaryTitle: "string",
  date: "string",
  diaryBody: "string",
  onSubmit: jest.fn(),
  onCancel: jest.fn(),
  onChange: jest.fn()
};

  