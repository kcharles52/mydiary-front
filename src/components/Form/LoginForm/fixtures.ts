export const LoginFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  