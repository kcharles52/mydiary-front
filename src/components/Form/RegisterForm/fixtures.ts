export const RegisterFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  