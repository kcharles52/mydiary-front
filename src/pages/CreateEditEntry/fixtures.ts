export const props = {
  Edit: false,
  title: "Form title",
  diaryTitle: "Test Diary Title",
  date: "20-2-19",
  diaryBody: "Test Diary body",
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  addEntryAction: jest.fn(),
  editEntryAction: jest.fn(),
  getEntryAction: jest.fn(() =>
    Promise.resolve({
      entry: {
        entry_id: 1,
        diaryTitle: "Title 1",
        diaryEntryBody:
          "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
        date: "12/12/2018",
        delete: jest.fn()
      }
    })
  ),
  match: { params: { entry_id: "1" } },
  editEntry: {
    entry: {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  },
  newEntry: {},
  history: { push: () => {} },
  location: { pathname: "/addEntry" }
};

export const entry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
