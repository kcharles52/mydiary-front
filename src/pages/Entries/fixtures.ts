export const props = {
  getEntriesAction: jest.fn(),
  deleteEntryAction: jest.fn(),
  entries: [
    {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      diaryTitle: "Title 2",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  deleted: [],
  handlesDelete: jest.fn(),
  history: {push:jest.fn()},
  location: { pathname: "/home" },
  match: { url: "/home" }
};

export const fetchedEntries =[
  {
    entry_id: 1,
    diaryTitle: "Title 1",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  },
  {
    entry_id: 2,
    diaryTitle: "Title 2",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  }
]