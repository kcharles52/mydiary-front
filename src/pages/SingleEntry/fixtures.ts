export const props = {
  getEntryAction: (id: string) => Promise.resolve(),
  deleteEntryAction: (id: string) => Promise.resolve(),
  fetchedEntry: { entry: {} },
  match: { params: { entry_id: "1" } },
  history: { push: jest.fn() },
  location: { pathname: "/Entry/1" }
};

export const fetchedEntry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
