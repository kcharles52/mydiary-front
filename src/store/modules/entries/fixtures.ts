export const intialState = {
  entries: [],
  entry: {},
  new: {},
  Message: "",
  deleted: []
};

export const fetchedEntries = {
  entries: [
    {
      entry_id: 1,
      title: "Title 1",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      title: "Title 2",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  Message:"Successfully fetched entries"
};

export const editedEntry = {
    entry_id: 1,
      title: "Edited",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
}