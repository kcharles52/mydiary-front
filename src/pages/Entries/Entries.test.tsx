//react libraries
import * as React from "react";

//third party libraries
import { mount, shallow } from "enzyme";

//fixtures
import { props, fetchedEntries } from "./fixtures";

//component
import { HomeLogedIn, mapStateToProps, mapDispatchToProps } from ".";

describe("Should render entries", () => {
  let wrapper, shallowWrapper;

  beforeEach(() => {
    wrapper = mount(<HomeLogedIn {...props} />);
    shallowWrapper = shallow(<HomeLogedIn {...props} />);
  });

  it("should render entries ", () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
  it("Rerender after deleting entries", () => {
    wrapper.setProps({ deleted: [{}] });
  });
  it("Should render title you have no entries", () => {
    wrapper.setProps({ entries: [] });
    expect(wrapper.instance().state.header).toEqual("You have no entries ");
  });

  it("should delete entry", () => {
    const spy = jest.spyOn(wrapper.instance(), "editRoute");
    wrapper.instance().forceUpdate();
    wrapper
      .find("button")
      .first()
      .simulate("click");
    expect(spy).toBeCalled();
    spy.mockRestore();
  });

  it("should redirect to edit entry page",()=>{
    const spy = jest.spyOn(wrapper.instance(), "handlesDelete");
    wrapper.instance().forceUpdate();
    wrapper
      .find("button")
      .last()
      .simulate("click");
    expect(spy).toBeCalled();
    spy.mockRestore();
  });
  
});

describe("Connect functions function", () => {
  describe("The mapStateToProps function", () => {
    it("should return the expected props object", () => {
      const state = {
        entries: { entries: fetchedEntries }
      };
      const props = mapStateToProps(state);
      expect(props.entries).toEqual(state.entries.entries);
    });
  });

  describe("The mapDispatchToProps function", () => {
    let dispatch, props;
    beforeEach(() => {
      dispatch = jest.fn();
      props = mapDispatchToProps(dispatch);
    });
    it("should dispatch getEntry when getEntryAction is called", () => {
      props.getEntriesAction();
      expect(dispatch).toHaveBeenCalled();
    });
    it("should dispatch getEntry when deleteEntryAction is called", () => {
      props.deleteEntryAction();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
