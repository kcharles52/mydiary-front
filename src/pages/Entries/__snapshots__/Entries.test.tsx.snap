// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render entries should render entries  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeLogedIn
    deleteEntryAction={[MockFunction]}
    deleted={Array []}
    entries={
      Array [
        Object {
          "date": "12/12/2018",
          "delete": [MockFunction],
          "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
          "diaryTitle": "Title 1",
          "entry_id": 1,
        },
        Object {
          "date": "12/12/2018",
          "delete": [MockFunction],
          "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
          "diaryTitle": "Title 2",
          "entry_id": 2,
        },
      ]
    }
    getEntriesAction={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    handlesDelete={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "pathname": "/home",
      }
    }
    match={
      Object {
        "url": "/home",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <div>
          <div
            className="entries"
          >
            <h1>
              Diary Entries
            </h1>
            <Entry
              date="12/12/2018"
              delete={[Function]}
              diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
              diaryTitle="Title 1"
              edit={[Function]}
              entry_id={1}
            />
            <Entry
              date="12/12/2018"
              delete={[Function]}
              diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
              diaryTitle="Title 2"
              edit={[Function]}
              entry_id={2}
            />
          </div>
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="entries"
          >
            <h1>
              Diary Entries
            </h1>
            <Entry
              date="12/12/2018"
              delete={[Function]}
              diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
              diaryTitle="Title 1"
              edit={[Function]}
              entry_id={1}
            />
            <Entry
              date="12/12/2018"
              delete={[Function]}
              diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
              diaryTitle="Title 2"
              edit={[Function]}
              entry_id={2}
            />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Diary Entries
              </h1>,
              Array [
                <Entry
                  date="12/12/2018"
                  delete={[Function]}
                  diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                  diaryTitle="Title 1"
                  edit={[Function]}
                  entry_id={1}
                />,
                <Entry
                  date="12/12/2018"
                  delete={[Function]}
                  diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                  diaryTitle="Title 2"
                  edit={[Function]}
                  entry_id={2}
                />,
              ],
            ],
            "className": "entries",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Diary Entries",
              },
              "ref": null,
              "rendered": "Diary Entries",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "date": "12/12/2018",
                "delete": [Function],
                "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                "diaryTitle": "Title 1",
                "edit": [Function],
                "entry_id": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "date": "12/12/2018",
                "delete": [Function],
                "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                "diaryTitle": "Title 2",
                "edit": [Function],
                "entry_id": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <div>
            <div
              className="entries"
            >
              <h1>
                Diary Entries
              </h1>
              <Entry
                date="12/12/2018"
                delete={[Function]}
                diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                diaryTitle="Title 1"
                edit={[Function]}
                entry_id={1}
              />
              <Entry
                date="12/12/2018"
                delete={[Function]}
                diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                diaryTitle="Title 2"
                edit={[Function]}
                entry_id={2}
              />
            </div>
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="entries"
            >
              <h1>
                Diary Entries
              </h1>
              <Entry
                date="12/12/2018"
                delete={[Function]}
                diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                diaryTitle="Title 1"
                edit={[Function]}
                entry_id={1}
              />
              <Entry
                date="12/12/2018"
                delete={[Function]}
                diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                diaryTitle="Title 2"
                edit={[Function]}
                entry_id={2}
              />
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Diary Entries
                </h1>,
                Array [
                  <Entry
                    date="12/12/2018"
                    delete={[Function]}
                    diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                    diaryTitle="Title 1"
                    edit={[Function]}
                    entry_id={1}
                  />,
                  <Entry
                    date="12/12/2018"
                    delete={[Function]}
                    diaryEntryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
                    diaryTitle="Title 2"
                    edit={[Function]}
                    entry_id={2}
                  />,
                ],
              ],
              "className": "entries",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Diary Entries",
                },
                "ref": null,
                "rendered": "Diary Entries",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "date": "12/12/2018",
                  "delete": [Function],
                  "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                  "diaryTitle": "Title 1",
                  "edit": [Function],
                  "entry_id": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {
                  "date": "12/12/2018",
                  "delete": [Function],
                  "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                  "diaryTitle": "Title 2",
                  "edit": [Function],
                  "entry_id": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
